src/frontends/tensorflow_common/src/op/gather.cpp

```diff
@@ -45,9 +63,39 @@ OutputVector translate_resource_gather_op(const NodeContext& node) {
 OutputVector translate_gather_v2_op(const NodeContext& node) {
     // GatherV2 has three inputs: data, indices, and axis by which data is sliced
     // batch_dims is an attribute and can vary
-    default_op_checks(node, 3, {"GatherV2"});
+    default_op_checks(node, 3, {"GatherV2"}, true);
+    auto params = node.get_input(0);
+    auto indices = node.get_input(1);
     auto axis = node.get_input(2);
     auto batch_dims = node.get_attribute<int64_t>("batch_dims", 0);
+    auto params_shape = make_shared<v3::ShapeOf>(params, ov::element::i32);
+    auto params_rank = make_shared<v3::ShapeOf>(params_shape, ov::element::i32);
+    auto complex_type_mark = as_type_ptr<ComplexTypeMark>(params.get_node_shared_ptr());
+
+    if (complex_type_mark) {
+        params = complex_type_mark->input_value(0);
+        auto zero = create_same_type_const_scalar<int32_t>(axis, 0);
+        // create a condition for the Select operation
+        auto condition = make_shared<v1::Less>(axis, zero);
+
+        // calculate the updated value for the axis
+        auto params_shape = make_shared<v3::ShapeOf>(params, ov::element::i32);
+        auto params_rank = make_shared<v3::ShapeOf>(params_shape, ov::element::i32);
+        auto updated_axis =
+            make_shared<v1::Subtract>(params_rank, make_shared<v0::Constant>(ov::element::i32, Shape{}, 1));
+
```

Comment:
please compute `updated_axis` as `axis - 1`. Then `ShapeOf` is not needed. For `batch_dims` apply the same logic